ios/AppLovinMAX.m

```diff
@@ -745,14 +757,23 @@ - (void)didLoadAd:(MAAd *)ad
     [self sendReactNativeEventWithName: name body: [self adInfoForAd: ad]];
 }
 
-- (void)handleNativeAdLoadFailureForAdUnitIdentifier:(NSString *)adUnitIdentifier error:(nullable MAError *)error
+- (void)sendReactNativeEventForAdLoadFailed:(NSString *)name forAdUnitIdentifier:(NSString *)adUnitIdentifier withError:(nullable MAError *)error
 {
-    [self sendReactNativeEventWithName: @"OnNativeAdLoadFailedEvent"
-                                  body: @{@"adUnitId" : adUnitIdentifier,
-                                          @"code" : @(error.code),
+    NSMutableDictionary *body = @{@"adUnitId": adUnitIdentifier}.mutableCopy;
+    
+    if ( error )
+    {
+        [body addEntriesFromDictionary: @{@"code" : @(error.code),
                                           @"message" : error.message,
                                           @"adLoadFailureInfo" : error.adLoadFailureInfo ?: @"",
                                           @"waterfall": [self createAdWaterfallInfo: error.waterfall]}];
+    }
+    else
+    {
+        [body addEntriesFromDictionary: @{@"code" : @(MAErrorCodeUnspecified)}];
+    }
+    
```

Comment:
We can consolidate this into one dictionary literaly expression. No need to create a dictinoary, then a mutable version, then another dictionary to be added into body`. That's 3 allocations in one call. We can keep it one allocation.

Reply:
Used the conditional operator to consolidate into one allocation.